sqlglot/optimizer/scope.py

```diff
@@ -438,7 +438,22 @@ def traverse(self):
         for child_scope in itertools.chain(
             self.cte_scopes, self.union_scopes, self.table_scopes, self.subquery_scopes
         ):
-            yield from child_scope.traverse()
+            if child_scope.is_union:
+                stack = [(False, child_scope)]
```

Comment:
We should use `deque`

Reply:
don’t you only need deque if you’re using a queue? this is just a stack

Reply:
Ah, yes, whoops